sieves/tasks/predictive/ner/bridges.py

```diff
@@ -0,0 +1,524 @@
+
+                    if entity_start_in_context >= 0:
+                        start = context_start + entity_start_in_context
+                        end = start + len(entity_text)
+
+                        # Create a new entity with start/end indices
+                        new_entity = Entity(text=doc_text[start:end], start=start, end=end, entity_type=entity_type)
+                        new_entities.append(new_entity)
+            else:
+                # If context approach fails, find all occurrences of the entity directly
+                entity_positions = []
+                pos = 0
+                while True:
+                    pos = doc_text_lower.find(entity_text_lower, pos)
+                    if pos == -1:
+                        break
+                    entity_positions.append(pos)
+                    pos += 1  # Move past the current position to find the next occurrence
+
+                # Create entities for each occurrence
+                for start in entity_positions:
+                    end = start + len(entity_text)
+                    new_entity = Entity(
+                        text=doc_text[start:end],  # Use the exact text from the document
+                        start=start,
+                        end=end,
+                        entity_type=entity_type,
+                    )
+                    new_entities.append(new_entity)
+        return new_entities
+
+    def integrate(self, results: Iterable[_BridgeResult], docs: Iterable[Doc]) -> Iterable[Doc]:
+        docs_list = list(docs)
+        results_list = list(results)
+
+        for doc, result in zip(docs_list, results_list):
+            # Create a new result with the same structure as the original
+            entities_with_position: list[Entity] = []
+
+            # Get the original text from the document
+            doc_text = doc.text if doc.text is not None else ""
+
+            # Handle result cases
+            if result is None:
+                # Skip if result is None
+                doc.results[self._task_id] = Entities(text=doc_text, entities=[])  # type: ignore
+            elif hasattr(result, "entities"):
+                # Process entities from result if available
+                entities_with_position = self.find_entity_positions(doc_text, result, entities_with_position)
+                # Create a new result with the updated entities
+                new_result = Entities(text=doc_text, entities=entities_with_position)
+                doc.results[self._task_id] = new_result
+            else:
+                # Default empty result
+                doc.results[self._task_id] = Entities(text=doc_text, entities=[])
+
+        return docs_list
+
+
+class DSPyNER(NERBridge[dspy_.PromptSignature, dspy_.Result, dspy_.InferenceMode]):
+    @property
+    def _prompt_template(self) -> str | None:
+        return None
+
+    @property
+    def _prompt_signature_description(self) -> str | None:
+        return """
```

Comment:
This describes the prompt signature, it's not a prompt itself. I.e. don't write this as instruction,  but as description of the prompt signature object/expected prompt result.